DropboxSync.UIL/BrokerEventListener.cs

```diff
@@ -60,27 +62,62 @@ public void Initialize()
                 host, port);
 
             Address address = new Address($"amqp://{username}:{password}@{host}:{port}");
-            AmqpConnection = new Connection(address);
+            
+            try
+            {
+                AmqpConnection = new Connection(address);
+                _logger.LogInformation("AMQP Connection established!");
+                AmqpConnection.Closed += Connection_Closed;
+            }
+            catch(Exception e)
+            {
+                ConnectionAttempts++;
 
-            _logger.LogInformation("AMQP Connection established!");
-            AmqpConnection.Closed += Connection_Closed;
+                _logger.LogError("{date} | Attempt {attempt} An error occured while trying to create connection : {ex}", 
+                    DateTime.Now, ConnectionAttempts, e.Message);
+                _logger.LogInformation("{date} | Trying to reconnect in 5 secondes", DateTime.Now);
+                Thread.Sleep(5000);
+                Initialize();
+            }
         }
 
         public void Start()
         {
             if (AmqpConnection is null) throw new NullReferenceException(nameof(AmqpConnection));
 
-            Session session = new Session(AmqpConnection);
-
-            ReceiverLink receiverLink = new ReceiverLink(session, "", _amqpCredentials.AmqpQueue);
-            receiverLink.Start(200, Message_Received);
-
-            _logger.LogInformation("{date} | Listening on AMQP", DateTime.Now);
+            try
+            {
+                Session session = new Session(AmqpConnection);
+                ReceiverLink receiverLink = new ReceiverLink(session, "", _amqpCredentials.AmqpQueue);
+                receiverLink.Start(200, Message_Received);
+                _logger.LogInformation("{date} | Listening on AMQP", DateTime.Now);
+            }
+            catch(AmqpException e)
+            {
+                _logger.LogError("{date} | Couldn't create AMQP Session : {ex}", DateTime.Now, e.Message);
+            }
+            catch(Exception e)
+            {
+                _logger.LogError("{date} | An error occured while trying to create a session to the broker : {ex}",
+                    DateTime.Now, e.Message);
+            }
         }
 
         private void Connection_Closed(IAmqpObject sender, Amqp.Framing.Error error)
         {
             _logger.LogCritical("Connection to the broker closed!");
+
+            if(ConnectionAttempts < 5)
```

Comment:
make it configurable with an env variable AMQP_MAX_CONNECTION_ATTEMPT or something like that

Reply:
no let's keep it like that for now